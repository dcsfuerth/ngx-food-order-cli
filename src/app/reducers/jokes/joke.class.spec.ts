import { Joke } from './joke.class';

describe('Joke', () => {
  describe('constructor', () => {
    let subject: Joke;

    describe('without params', () => {
      beforeEach(() => {
        subject = new Joke();
      });

      it('uses the default params', () => {
        expect(subject).toMatchSnapshot();
      });
    });

    describe('with params', () => {
      beforeEach(() => {
        subject = new Joke({ id: '42', joke: 'Knock, knock!' });
      });

      it('uses the given params', () => {
        expect(subject).toMatchSnapshot();
      });
    });
  });
});
