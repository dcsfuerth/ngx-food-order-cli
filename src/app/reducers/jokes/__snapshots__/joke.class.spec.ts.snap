// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Joke constructor with params uses the given params 1`] = `"{\\"id\\":\\"42\\",\\"joke\\":\\"Knock, knock!\\",\\"categories\\":[\\"nerdy\\"]}"`;

exports[`Joke constructor without params uses the default params 1`] = `"{\\"id\\":\\"\\",\\"joke\\":\\"\\",\\"categories\\":[\\"nerdy\\"]}"`;
